### PR TITLE
added ev43 to support fast pulse rates

### PR DESCRIPTION
**Pull request by geishm-ansto**
### Description of Work

Added ev43 schema to support instruments with fast pulse rates that allow for more efficient data transfer. 

### Issue

No documented issue but comes about from framework evaluation at Ansto.

### Developer Checklist

- [ ] If there are new schema in this PR I have added them to the list in README.md
- [ ] If there are breaking changes to a schema, I have used a new file identifier and updated the list in README.md
- [ ] There is some documentation here or in the flat buffer file on the use case for this data, including which component is intended to send the data and/or which is the intended receiver.

*A file identifier can be generated [here](https://www.random.org/strings/?num=1&len=4&digits=on&upperalpha=on&loweralpha=on&unique=on&format=html&rnd=new)*

## Approval Criteria

This PR should not be merged until Tobias R, Mark K and Jack H have given their explicit approval in the comments section.

### CODE REVIEW

**Review by mortenjc — COMMENTED** [review]
I am generally ok with this PR although this is not something I normally work with.

Not sure though why there is a need to add a new array definition to the is84 schema?

**Comment by geishm-ansto** [comment]
@mortenjc The ev42 message for a single pulse time included the ISIS data so, in principle, the message could be updated for each pulse. I extended the message type to support that capability if needed. The choice of ArrayISISData rather than an array of ISISData came about because I didn't think that the run-state may change which each pulse. The facility data is not needed for us at Ansto but I just tried to keep it consistent with ev42, however I am open to suggestions.

File: README.md [hunk 1]
```diff
@@ -61,6 +61,7 @@ f141        f141_ntarraydouble.fbs            [OBSOLETE] A simple array of doubl
 f142        f142_logdata.fbs                  For log data, for example forwarded EPICS PV update
 f143        f143_structure.fbs                [OBSOLETE] Arbitrary nested data
 ev42        ev42_events.fbs                   Multi-institution neutron event data
+ev43        ev43_events.fbs                   Multi-institution neutron event data to support fast pulse rates
```
Comment (zjttoefs) [inline README.md:64]: I'd make that
`Multi-institution neutron event data from multiple pulses`
We could even specify that ev42 only contains a single pulse.

**Review by zjttoefs — COMMENTED** [review]
I'd be inclined to drop the facility_specific_data for now. It is backward compatible to add stuff later, if someone decided they need it. Then we don't need to guess what ISIS would like to add in is84 and keep that as is. I haven't checked if the AdcPulseDebug makes sense unchanged and we don't have to.
I'd like to have a statement on the length of the arrays (if present): `pulse_time` and `pulse_index` are the same length (per pulse), while `time_of_flight` and `detector_id` are the same per event.

File: README.md (hunk 1 above)
Comment (geishm-ansto) [reply README.md:64]: That makes sense. I'll drop the facility data and change the READ.me description.

**Comment by geishm-ansto** [comment]
@zjttoefs Are there any other changes you would like to see?

**Comment by zjttoefs** [comment]
No @geishm-ansto, that looks very good now. I'll just merge with a heads up to @mkoennecke just for information.

**Pull request merged**